batchflow/models/validator/validator.py

```diff
@@ -0,0 +1,337 @@
+        """
+        pass
+
+    def compute_metrics(self, targets, predictions, *metrics, **metric_configs):
+        """ Metrics computation. """
+        metrics = list(set([*metrics, *metric_configs.keys()]))
+        values = {}
+        for _metric in metrics:
+            if hasattr(self, _metric):
+                values[_metric] = getattr(self, _metric)(targets, predictions)
+            else:
+                metric_config = metric_configs.get(_metric, {})
+                metric_class = metric_config.pop('class', 'classification')
+                evaluate = metric_config.pop('evaluate', {})
+                metrics = METRICS[metric_class](targets, predictions, **metric_config)
+                values[_metric] = metrics.evaluate(_metric, **evaluate)
+        return values
+
+    @classmethod
+    def check_api(cls, methods=('train', 'inference'), warning=True, exception=False):
+        """ Check that Validator child class implements necessary methods.
+
+        Parameters
+        ----------
+        methods : list
+            list of methods to check.
+        warning : bool
+            if True, call warning.
+        exception : bool
+            if True, raise exception.
+
+        Returns
+        -------
+        error: bool
+        """
+        error = False
+        if isinstance(methods, str):
+            methods = (methods, )
+        if exception:
+            def _warning(msg):
+                raise NotImplementedError(msg)
+        elif warning:
+            _warning = warnings.warn
+        else:
+            _warning = lambda msg: msg
+        for meth in methods:
+            cond = all([_get_class_that_defined_method(getattr(cls, _meth)) == Validator for _meth in meth.split('|')])
+            if cond:
+                error = True
+                _ = _warning('Method "{}" is not implemented in class {}'.format(meth, cls.__class__))
+        return error
+
+    def check_config(self, keys=('train', 'test'), warning=True, exception=False):
+        """ Check that config has necessary keys.
+
+        Parameters
+        ----------
+        keys : list
+            list of keys to check.
+        warning : bool
+            if True, call warning. If exception
+        exception : bool
+            if True, raise exception.
+
+        Returns
+        -------
+        error: bool
+        """
+        error = False
+        if isinstance(keys, str):
+            keys = (keys, )
+        if exception:
+            def _warning(msg):
+                raise NotImplementedError(msg)
+        elif warning:
+            _warning = warnings.warn
+        else:
+            _warning = lambda msg: msg
+        for key in keys:
+            cond = all([_key not in self.config for _key in key.split('|')])
+            if cond:
+                error = True
+                _ = _warning('Key "{}" was not founded in config: {}'.format(key, self.config))
+        return error
+
+    def run(self):
```

Comment:
I would rather move `run` (and `_run`/`_run_cv`) higher, as most of the other methods (like train/test/inference) are mere ~`abstract` placeholders